InnerEye/ML/deep_learning_config.py

```diff
@@ -183,39 +183,184 @@ def add_subfolder(self, subfolder: str) -> DeepLearningFileSystemConfig:
+    perform_training_set_inference: bool = \
+        param.Boolean(False,
+                      doc="If False (default), run full image inference on validation and test set after training. If "
+                          "True, also run full image inference on the training set")
+    perform_validation_and_test_set_inference: bool = \
+        param.Boolean(True,
+                      doc="If True (default), run full image inference on validation and test set after training.")
+    weights_url: str = param.String(doc="If provided, a url from which weights will be downloaded and used for model "
+                                        "initialization.")
+    local_weights_path: Optional[Path] = param.ClassSelector(class_=Path,
+                                                             default=None,
+                                                             allow_None=True,
+                                                             doc="The path to the weights to use for model "
+                                                                 "initialization, when training outside AzureML.")
+    generate_report: bool = param.Boolean(default=True,
+                                          doc="If True (default), write a modelling report in HTML format. If False,"
+                                              "do not write that report.")
+    # The default multiprocessing start_method in both PyTorch and the Python standard library is "fork" for Linux and
+    # "spawn" (the only available method) for Windows. There is some evidence that using "forkserver" on Linux
+    # can reduce the chance of stuck jobs.
+    multiprocessing_start_method: MultiprocessingStartMethod = \
+        param.ClassSelector(class_=MultiprocessingStartMethod,
+                            default=(MultiprocessingStartMethod.spawn if is_windows()
+                                     else MultiprocessingStartMethod.fork),
+                            doc="Method to be used to start child processes in pytorch. Should be one of forkserver, "
+                                "fork or spawn. If not specified, fork is used on Linux and spawn on Windows. "
+                                "Set to forkserver as a possible remedy for stuck jobs.")
+    monitoring_interval_seconds: int = param.Integer(0, doc="Seconds delay between logging GPU/CPU resource "
+                                                            "statistics. If 0 or less, do not log any resource "
+                                                            "statistics.")
+
+    def validate(self) -> None:
+        if self.weights_url and self.local_weights_path:
+            raise ValueError("Cannot specify both local_weights_path and weights_url.")
+
+        if self.number_of_cross_validation_splits == 1:
+            raise ValueError(f"At least two splits required to perform cross validation found "
```

Comment:
I know this doc is not new but could we add here to the error message "If you want to train a single training job without cross-validation, specify number_of_cross_validation_splits=0"?

Reply:
good suggestion, thanks